src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -198,19 +220,16 @@ export class WorkspaceSavedObjectsClientWrapper {
       objects: Array<SavedObjectsBulkUpdateObject<T>>,
       options?: SavedObjectsBulkUpdateOptions
     ): Promise<SavedObjectsBulkUpdateResponse<T>> => {
-      const workspaceIds = objects.reduce<string[]>((acc, cur) => {
-        if (this.isRelatedToWorkspace(cur.type)) {
-          acc.push(cur.id);
+      for (const object of objects) {
+        const permitted = await validateUpdateWithWorkspacePermission(
```

Comment:
I think a least we can `batchGet` all the `objects` here instead of `get` one by one in `validateUpdateWithWorkspacePermission`.

Reply:
Similar to `bulkGetWithWorkspacePermissionControl`